src/provider/mod.rs

```diff
@@ -87,23 +88,23 @@ impl StackableProvider {
     }
 
     fn get_package(pod: &Pod) -> Result<Package, StackableError> {
-        let containers = pod.containers();
-        return if containers.len().ne(&1) {
-            let e = PodValidationError {
-                msg: String::from("Size of containers list in PodSpec has to be exactly 1"),
-            };
-            Err(e)
+        if let Some((container, [])) = pod.containers().split_first() {
+            container
+                .image()
+                .and_then(|maybe_ref| maybe_ref.ok_or(anyhow!("Image is required.")))
+                .and_then(Package::try_from)
+                .map_err(|err| PodValidationError {
+                    msg: format!(
+                        r#"Unable to get package reference from pod "{}": {}"#,
```

Comment:
We have a kinda looseish convention of wrapping values in square brackets:

```suggestion
                        "Unable to get package reference from pod [{}]: {}",
```

Reply:
Convention for error messages applied